hpx/parallel/segmented_algorithms/find.hpp

```diff
@@ -0,0 +1,357 @@
+                }
+
+                // handle all of the full partitions
+                for (++sit; sit != send; ++sit)
+                {
+                    beg = traits::begin(sit);
+                    end = traits::end(sit);
+                    if (beg != end)
+                    {
+                        segments.push_back(
+                            hpx::make_future<InIter>(
+                                dispatch_async(traits::get_id(sit), algo,
+                                    policy, forced_seq(), beg, end, f_or_val),
+                                [sit,end,last](local_iterator_type const& out)
+                                    -> InIter
+                                {
+                                    if(out != end)
+                                        return traits::compose(sit, out);
+                                    else
+                                        return last;
+                                }));
+                    }
+                }
+
+                // handle the beginning of the last partition
+                beg = traits::begin(sit);
+                end = traits::local(last);
+                if (beg != end)
+                {
+                    segments.push_back(
+                        hpx::make_future<InIter>(
+                            dispatch_async(traits::get_id(sit), algo,
+                                policy, forced_seq(), beg, end, f_or_val),
+                            [sit,end,last](local_iterator_type const& out)
+                                -> InIter
+                            {
+                                if(out != end)
+                                    return traits::compose(sit, out);
+                                else
+                                    return last;
+                            }));
+                }
+            }
+            return result::get(
+                dataflow(
+                    [=](std::vector<hpx::future<InIter> > && r)
+                        ->  InIter
+                    {
+                        // handle any remote exceptions, will throw on error
+                        std::list<std::exception_ptr> errors;
+                        parallel::util::detail::handle_remote_exceptions<
+                            ExPolicy
+                        >::call(r, errors);
+
+                        std::vector<InIter> res =
+                            hpx::util::unwrapped(std::move(r));
+                        auto it = res.begin();
+                        while(it!=res.end())
+                        {
+                            if(*it != last)
+                                return *it;
+                            it++;
+                        }
+                        return res.back();
+                    },
+                    std::move(segments)));
+        }
+
+        template <typename ExPolicy, typename InIter, typename T>
+        inline typename std::enable_if<
```

Comment:
In this function and all others - I think it's redundant to verify the execution policy everywhere. @hkaiser, what do you think?